### DISCUSSION TITLE
No install of SBFspot on 2023-12-05-raspios-bullseye-armhf-lite.img.xz

### DISCUSSION THREAD

**Discussion by Rammy61** [SBFspot/sbfspot-config#72] [General]
I want to install SBFspot on a RPiZero-W but the install process is stopped because the software thinks I am using a wrong image.

![Screenshot_2024-01-31_19-16-37](https://github.com/SBFspot/sbfspot-config/assets/35842109/41e8b13d-3350-429c-87fe-efbcd41e9f35)

When I change the line:

`    if [[ ! "Raspbian Debian" =~ "\<$lsb_id\>" ]]; then`

in

	`if [[ ! "Raspbian" =~ "$lsb_id" ]]; then`

and start the process again it works.

Regards.

Rammy

**Comment by sbf-** [comment]
Don't know what's happening on your system, but this piece of code works as intended;

```
pi@raspi32:~ $ re_id="\<Debian\>"
pi@raspi32:~ $ if [[ ! "Raspbian Debian" =~ $re_id ]]; then echo "NOK"; else echo "OK"; fi
OK
pi@raspi32:~ $ re_id="\<Raspbian\>"
pi@raspi32:~ $ if [[ ! "Raspbian Debian" =~ $re_id ]]; then echo "NOK"; else echo "OK"; fi
OK
pi@raspi32:~ $ re_id="\<Unknown\>"
pi@raspi32:~ $ if [[ ! "Raspbian Debian" =~ $re_id ]]; then echo "NOK"; else echo "OK"; fi
NOK
```

However, this doesn't:
```
pi@raspi32:~ $ if [[ ! "Raspbian Debian" =~ "\<Raspbian\>" ]]; then echo "NOK"; else echo "OK"; fi
NOK
```

**Reply by Rammy61** [reply]
You want to check if the literal text of $lsb_id is contained in "Raspbian Debian" string.
For that you need to inclose the text in "" or ''.

The bash manual says:
If you quote any part of the pattern, using any of the shell’s quoting mechanisms, the quoted portion is matched literally. This means every character in the quoted portion matches itself, instead of having any special pattern matching meaning. 
See, https://www.gnu.org/software/bash/manual/bash.html#index-_005b_005b.

So if you change the line in:

`if [[ ! "Raspbian Debian" =~ "$lsb_id" ]]; then`

it works as expected.
However if you want to use the '/<' and '/>' as part of a regular expression then it should be places outside the string to test.
The code line would then look like:

`if [[ ! "Raspbian Debian" =~ \<"$lsb_id"\> ]]; then`

**Reply by sbf-** [reply]
> So if you change the line in:
> if [[ ! "Raspbian Debian" =~ "$lsb_id" ]]; then
> it works as expected.

That' not completely true, because when `lsb_id` is empty ("") for some reason, the test will pass. It will also pass with a substring and that's not wat I want:
```
pi@raspi32:~ $ lsb_id=""
pi@raspi32:~ $ if [[ ! "Raspbian Debian" =~ "$lsb_id" ]]; then echo "NOK"; else echo "OK"; fi
OK
pi@raspi32:~ $ lsb_id="Rasp"
pi@raspi32:~ $ if [[ ! "Raspbian Debian" =~ "$lsb_id" ]]; then echo "NOK"; else echo "OK"; fi
OK
```

**Reply by Rammy61** [reply]
In case of an empty command:

`lsb_id=$(lsb_release -si)`

That could be replaced for:

`lsb_id=$(lsb_release -si); [[ -z "$lsb_release" ]] && { echo "Can't determine image name"; exit 1; }`

For the substring you could use:

`[[ -z "$( echo "Raspian Debian" | grep -Eow "$lsb_id" )" ]] &&  { echo "Error: Not running Raspian nor Raspberry Pi OS."; exit 1; }`

With the latter construction you don't need to test for an empty $lsb_id.

**Comment by sbf-** [comment]
I burned the same image and ran it on an old RPi1b without issues...

![image](https://github.com/SBFspot/sbfspot-config/assets/1931158/2c84769b-4069-4b51-90a9-3fd5401beb64)

**Comment by Rammy61** [comment]
Did you bring the image up to date before starting setting-up SBFspot?

**Reply by sbf-** [reply]
Yes, I executed `sudo apt full-upgrade`

**Reply by Rammy61** [reply]
Today I spent the whole day to try to reproduce the issue on a RPi1-B+_v1.2 and on my RPi0-W_v1.1, I did not succeed.
So I can't tell you why it happend the first time with the RPi0-W.
If it happens again I shall report it to you.

For now thanks that you took the time to investigate the issue further.
